packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx

```diff
@@ -1,48 +1,44 @@
 import { css, keyframes } from '@emotion/css';
-import React from 'react';
-
-import { GrafanaTheme2 } from '@grafana/data';
+import React, { CSSProperties } from 'react';
 
 import { useStyles2 } from '../../themes';
 
 export interface LoadingBarProps {
-  width?: string;
-  height?: string;
+  width: number;
   ariaLabel?: string;
 }
 
-export function LoadingBar({ width, height, ariaLabel = 'Loading bar' }: LoadingBarProps) {
-  const styles = useStyles2(getStyles(width, height));
+const MILLISECONDS_PER_PIXEL = 2.4;
+
+export function LoadingBar({ width, ariaLabel = 'Loading bar' }: LoadingBarProps) {
+  const styles = useStyles2(getStyles);
+  const animationSpeed = Math.min(Math.max(Math.round(width * MILLISECONDS_PER_PIXEL), 500), 4000);
```

Comment:
Nit: I think by extracting the min and max timing values to constants as well we could make the code easier to read.

```javascript
const MILLISECONDS_PER_PIXEL = 2.4;
const MIN_SPEED = 500; // ms
const MAX_SPEED = 4000; // ms
```

```suggestion
  const animationSpeed = Math.min(Math.max(Math.round(width * MILLISECONDS_PER_PIXEL), MIN_SPEED), MAX_SPEED);
```

Reply:
done!